<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MOJITOO">
<title>Using MOJITOO in R for ArchR • MOJITOO</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Using MOJITOO in R for ArchR">
<meta property="og:description" content="MOJITOO">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MOJITOO</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ArchRObject_integration.html">Using MOJITOO in R for ArchR</a>
    <a class="dropdown-item" href="../articles/Matrix_integration.html">Using MOJITOO in R for Matrix input</a>
    <a class="dropdown-item" href="../articles/SeuratObject_integration.html">Using MOJITOO in R for Seurat</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Using MOJITOO in R for ArchR</h1>
                        <h4 data-toc-skip class="author">Mingbo
Cheng</h4>
            
            <h4 data-toc-skip class="date">October 23, 2024</h4>
      
      
      <div class="d-none name"><code>ArchRObject_integration.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="download-data">1. download data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h3>
<pre><code><span class="co">## 2024-10-23 12:17:16 URL:https://cf.10xgenomics.com/samples/cell-arc/1.0.0/pbmc_granulocyte_sorted_10k/pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz [2051027831/2051027831] -&gt; "pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz" [1]</span>
<span class="co">## 2024-10-23 12:17:16 URL:https://cf.10xgenomics.com/samples/cell-arc/1.0.0/pbmc_granulocyte_sorted_10k/pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz.tbi [1027204/1027204] -&gt; "pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz.tbi" [1]</span>
<span class="co">## WARNING: cannot verify costalab.ukaachen.de's certificate, issued by ‘CN=GEANT OV RSA CA 4,O=GEANT Vereniging,C=NL’:</span>
<span class="co">##   Unable to locally verify the issuer's authority.</span>
<span class="co">## 2024-10-23 12:17:16 URL:https://costalab.ukaachen.de/open_data/MOJITOO/PBMC-Multiom_annotation.tsv [400178/400178] -&gt; "PBMC-Multiom_annotation.tsv" [1]</span>
<span class="co">## 2024-10-23 12:17:18 URL:https://cf.10xgenomics.com/samples/cell-arc/1.0.0/pbmc_granulocyte_sorted_10k/pbmc_granulocyte_sorted_10k_filtered_feature_bc_matrix.h5 [162282142/162282142] -&gt; "pbmc_granulocyte_sorted_10k_filtered_feature_bc_matrix.h5" [1]</span></code></pre>
</div>
<div class="section level3">
<h3 id="library">2. library<a class="anchor" aria-label="anchor" href="#library"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MOJITOO</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">parallel</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ArchR</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: package 'BiocGenerics' was built under R version 4.1.1</span></code></pre>
<pre><code><span class="co">## Warning: package 'S4Vectors' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## Warning: package 'IRanges' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## Warning: package 'GenomeInfoDb' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## Warning: package 'Biobase' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## Warning: package 'data.table' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## Warning: package 'Matrix' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## Warning: package 'magrittr' was built under R version 4.1.3</span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nanx.me/ggsci/" class="external-link">ggsci</a></span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://stringr.tidyverse.org" class="external-link">stringr</a></span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: package 'stringr' was built under R version 4.1.3</span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v86</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: package 'AnnotationDbi' was built under R version 4.1.3</span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ArchR/man/addArchRGenome.html" class="external-link">addArchRGenome</a></span><span class="op">(</span><span class="st">"hg38"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Setting default genome to Hg38.</span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ArchR/man/addArchRThreads.html" class="external-link">addArchRThreads</a></span><span class="op">(</span><span class="fl">30</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Setting default number of Parallel threads to 30.</span></code></pre>
</div>
<div class="section level3">
<h3 id="create-archr-proj">3. Create ArchR proj<a class="anchor" aria-label="anchor" href="#create-archr-proj"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">inputFiles</span> <span class="op">&lt;-</span> <span class="st">"pbmc_granulocyte_sorted_10k_atac_fragments.tsv.gz"</span>
<span class="va">ArrowFiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ArchR/man/createArrowFiles.html" class="external-link">createArrowFiles</a></span><span class="op">(</span>inputFiles <span class="op">=</span> <span class="va">inputFiles</span>,
                               sampleNames <span class="op">=</span> <span class="st">"pbmc"</span>,
                               minTSS <span class="op">=</span> <span class="fl">0</span>, <span class="co">##4</span>
                               filterFrags <span class="op">=</span> <span class="fl">0</span>, <span class="co">## 1000</span>
                               addTileMat <span class="op">=</span> <span class="cn">T</span>,
                               addGeneScoreMat <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## filterFrags is no longer a valid input. Please use minFrags! Setting filterFrags value to minFrags!</span></code></pre>
<pre><code><span class="co">## Using GeneAnnotation set by addArchRGenome(Hg38)!</span>
<span class="co">## Using GeneAnnotation set by addArchRGenome(Hg38)!</span></code></pre>
<pre><code><span class="co">## Warning: package 'Biostrings' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## Warning: package 'XVector' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## ArchR logging to : ArchRLogs/ArchR-createArrows-fa26e1254d494-Date-2024-10-23_Time-12-17-47.log</span>
<span class="co">## If there is an issue, please report to github with logFile!</span></code></pre>
<pre><code><span class="co">## Cleaning Temporary Files</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:17:47 : Batch Execution w/ safelapply!, 0 mins elapsed.</span></code></pre>
<pre><code><span class="co">## (pbmc : 1 of 1) Determining Arrow Method to use!</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:17:47 : (pbmc : 1 of 1) Reading In Fragments from inputFiles (readMethod = tabix), 0.001 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:17:47 : (pbmc : 1 of 1) Tabix Bed To Temporary File, 0.001 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:18:57 : (pbmc : 1 of 1) Successful creation of Temporary File, 1.174 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:18:57 : (pbmc : 1 of 1) Creating ArrowFile From Temporary File, 1.174 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:20:04 : (pbmc : 1 of 1) Successful creation of Arrow File, 2.296 mins elapsed.</span></code></pre>
<pre><code><span class="co">## Warning: package 'gridExtra' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:21:21 : (pbmc : 1 of 1) CellStats : Number of Cells Pass Filter = 639351 , 3.578 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:21:21 : (pbmc : 1 of 1) CellStats : Median Frags = 5 , 3.578 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:21:21 : (pbmc : 1 of 1) CellStats : Median TSS Enrichment = 0.099 , 3.579 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:21:50 : (pbmc : 1 of 1) Adding Additional Feature Counts!, 4.055 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:22:15 : (pbmc : 1 of 1) Removing Fragments from Filtered Cells, 4.474 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:22:15 : (pbmc : 1 of 1) Creating Filtered Arrow File, 4.474 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:23:23 : (pbmc : 1 of 1) Finished Constructing Filtered Arrow File!, 5.606 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:23:26 : (pbmc : 1 of 1) Adding TileMatrix!, 5.653 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:26:27 : (pbmc : 1 of 1) Finished Creating Arrow File, 8.678 mins elapsed.</span></code></pre>
<pre><code><span class="co">## ArchR logging successful to : ArchRLogs/ArchR-createArrows-fa26e1254d494-Date-2024-10-23_Time-12-17-47.log</span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ArrowFiles</span> <span class="op">&lt;-</span> <span class="st">"pbmc.arrow"</span>
<span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ArchR/man/ArchRProject.html" class="external-link">ArchRProject</a></span><span class="op">(</span>ArrowFiles <span class="op">=</span> <span class="va">ArrowFiles</span>, outputDirectory <span class="op">=</span> <span class="st">"PBMC"</span>, copyArrows <span class="op">=</span> <span class="cn">T</span>, showLogo<span class="op">=</span><span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Using GeneAnnotation set by addArchRGenome(Hg38)!</span></code></pre>
<pre><code><span class="co">## Using GeneAnnotation set by addArchRGenome(Hg38)!</span></code></pre>
<pre><code><span class="co">## Validating Arrows...</span></code></pre>
<pre><code><span class="co">## Getting SampleNames...</span></code></pre>
<pre><code><span class="co">## 1 </span>
<span class="co">## Copying ArrowFiles to Ouptut Directory! If you want to save disk space set copyArrows = FALSE</span>
<span class="co">## 1 </span>
<span class="co">## Getting Cell Metadata...</span>
<span class="co">## 1 </span>
<span class="co">## Merging Cell Metadata...</span>
<span class="co">## Initializing ArchRProject...</span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"PBMC-Multiom_annotation.tsv"</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pbmc#"</span>, <span class="op">(</span><span class="va">meta</span><span class="op">$</span><span class="va">barcode</span><span class="op">)</span><span class="op">)</span>

<span class="va">bc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">proj</span><span class="op">@</span><span class="va">cellColData</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span><span class="op">)</span>
<span class="va">meta</span> <span class="op">&lt;-</span> <span class="va">meta</span><span class="op">[</span><span class="va">bc</span>, <span class="op">]</span>
<span class="va">proj</span> <span class="op">&lt;-</span> <span class="va">proj</span><span class="op">[</span><span class="va">bc</span>, <span class="op">]</span>
<span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/setCellCol.html">setCellCol</a></span><span class="op">(</span><span class="va">proj</span>, meta<span class="op">=</span><span class="va">meta</span><span class="op">)</span>


<span class="va">gene.coords</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-exonsBy.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v86</span>, filter <span class="op">=</span> <span class="op">~</span> <span class="va">gene_biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span><span class="op">)</span>
<span class="va">ucsc.levels</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace</a></span><span class="op">(</span>string<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"chr"</span>,<span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gene.coords</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">""</span><span class="op">)</span>, pattern<span class="op">=</span><span class="st">"chrMT"</span>, replacement<span class="op">=</span><span class="st">"chrM"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">gene.coords</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">ucsc.levels</span>
<span class="va">genebody.coords</span> <span class="op">&lt;-</span> <span class="fu">keepStandardChromosomes</span><span class="op">(</span><span class="va">gene.coords</span>, pruning.mode <span class="op">=</span> <span class="st">'coarse'</span><span class="op">)</span>

<span class="va">mtxs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X_h5.html" class="external-link">Read10X_h5</a></span><span class="op">(</span><span class="st">"pbmc_granulocyte_sorted_10k_filtered_feature_bc_matrix.h5"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Genome matrix has multiple modalities, returning a list of matrices for this genome</span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">RNA</span> <span class="op">&lt;-</span> <span class="va">mtxs</span><span class="op">[[</span><span class="st">"Gene Expression"</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">RNA</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"pbmc#"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">RNA</span><span class="op">)</span><span class="op">)</span>
<span class="va">RNA</span> <span class="op">&lt;-</span> <span class="va">RNA</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">proj</span><span class="op">@</span><span class="va">cellColData</span><span class="op">)</span><span class="op">]</span>

<span class="va">inter_gene_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">RNA</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">elementMetadata</a></span><span class="op">(</span><span class="va">genebody.coords</span><span class="op">)</span><span class="op">$</span><span class="va">gene_name</span><span class="op">)</span>

<span class="va">gtfMatch</span> <span class="op">&lt;-</span> <span class="va">genebody.coords</span><span class="op">[</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/na.omit.data.table.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-comparison.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">RNA</span><span class="op">)</span>, <span class="va">genebody.coords</span><span class="op">$</span><span class="va">gene_name</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gtfMatch</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">gtfMatch</span><span class="op">$</span><span class="va">gene_name</span>


<span class="va">seRNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></span><span class="op">(</span>
  assays <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/SimpleList-class.html" class="external-link">SimpleList</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">RNA</span><span class="op">[</span><span class="va">inter_gene_name</span>, <span class="op">]</span><span class="op">)</span>,
  rowData <span class="op">=</span> <span class="va">gtfMatch</span>
<span class="op">)</span>
<span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ArchR/man/addGeneExpressionMatrix.html" class="external-link">addGeneExpressionMatrix</a></span><span class="op">(</span><span class="va">proj</span>, <span class="va">seRNA</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ArchR logging to : ArchRLogs/ArchR-addGeneExpressionMatrix-fa26e74ac0a62-Date-2024-10-23_Time-12-26-57.log</span>
<span class="co">## If there is an issue, please report to github with logFile!</span>
<span class="co">## Overlap w/ scATAC = 1</span>
<span class="co">## 2024-10-23 12:26:58 : </span>
<span class="co">## Overlap Per Sample w/ scATAC : pbmc=11763</span>
<span class="co">## 2024-10-23 12:26:58 : </span>
<span class="co">## 2024-10-23 12:27:00 : Batch Execution w/ safelapply!, 0 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:02 : Adding pbmc to GeneExpressionMatrix for Chr (1 of 23)!, 0.029 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:05 : Adding pbmc to GeneExpressionMatrix for Chr (2 of 23)!, 0.07 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:07 : Adding pbmc to GeneExpressionMatrix for Chr (3 of 23)!, 0.108 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:10 : Adding pbmc to GeneExpressionMatrix for Chr (4 of 23)!, 0.161 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:12 : Adding pbmc to GeneExpressionMatrix for Chr (5 of 23)!, 0.197 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:14 : Adding pbmc to GeneExpressionMatrix for Chr (6 of 23)!, 0.233 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:17 : Adding pbmc to GeneExpressionMatrix for Chr (7 of 23)!, 0.286 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:20 : Adding pbmc to GeneExpressionMatrix for Chr (8 of 23)!, 0.322 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:22 : Adding pbmc to GeneExpressionMatrix for Chr (9 of 23)!, 0.358 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:25 : Adding pbmc to GeneExpressionMatrix for Chr (10 of 23)!, 0.41 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:27 : Adding pbmc to GeneExpressionMatrix for Chr (11 of 23)!, 0.446 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:29 : Adding pbmc to GeneExpressionMatrix for Chr (12 of 23)!, 0.483 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:32 : Adding pbmc to GeneExpressionMatrix for Chr (13 of 23)!, 0.535 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:35 : Adding pbmc to GeneExpressionMatrix for Chr (14 of 23)!, 0.57 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:37 : Adding pbmc to GeneExpressionMatrix for Chr (15 of 23)!, 0.606 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:40 : Adding pbmc to GeneExpressionMatrix for Chr (16 of 23)!, 0.658 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:42 : Adding pbmc to GeneExpressionMatrix for Chr (17 of 23)!, 0.694 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:44 : Adding pbmc to GeneExpressionMatrix for Chr (18 of 23)!, 0.731 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:47 : Adding pbmc to GeneExpressionMatrix for Chr (19 of 23)!, 0.782 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:49 : Adding pbmc to GeneExpressionMatrix for Chr (20 of 23)!, 0.819 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:52 : Adding pbmc to GeneExpressionMatrix for Chr (21 of 23)!, 0.855 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:55 : Adding pbmc to GeneExpressionMatrix for Chr (22 of 23)!, 0.905 mins elapsed.</span>
<span class="co">## 2024-10-23 12:27:57 : Adding pbmc to GeneExpressionMatrix for Chr (23 of 23)!, 0.941 mins elapsed.</span>
<span class="co">## ArchR logging successful to : ArchRLogs/ArchR-addGeneExpressionMatrix-fa26e74ac0a62-Date-2024-10-23_Time-12-26-57.log</span></code></pre>
</div>
<div class="section level3">
<h3 id="rna-atac-dimension-reductions">4. RNA &amp; ATAC dimension reductions<a class="anchor" aria-label="anchor" href="#rna-atac-dimension-reductions"></a>
</h3>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ArchR/man/addIterativeLSI.html" class="external-link">addIterativeLSI</a></span><span class="op">(</span>ArchRProj <span class="op">=</span> <span class="va">proj</span>,
                        useMatrix <span class="op">=</span> <span class="st">"TileMatrix"</span>,
                        name <span class="op">=</span> <span class="st">"IterativeLSI"</span>,
                        iterations <span class="op">=</span> <span class="fl">2</span>,
                        varFeatures <span class="op">=</span> <span class="fl">25000</span>,
                        dimsToUse <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Checking Inputs...</span></code></pre>
<pre><code><span class="co">## ArchR logging to : ArchRLogs/ArchR-addIterativeLSI-fa26e48c33538-Date-2024-10-23_Time-12-28-01.log</span>
<span class="co">## If there is an issue, please report to github with logFile!</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:28:01 : Computing Total Across All Features, 0 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:28:02 : Computing Top Features, 0.016 mins elapsed.</span></code></pre>
<pre><code><span class="co">## ###########</span>
<span class="co">## 2024-10-23 12:28:04 : Running LSI (1 of 2) on Top Features, 0.046 mins elapsed.</span>
<span class="co">## ###########</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:28:04 : Sampling Cells (N = 10000) for Estimated LSI, 0.047 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:28:04 : Creating Sampled Partial Matrix, 0.047 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:28:33 : Computing Estimated LSI (projectAll = FALSE), 0.523 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:29:47 : Identifying Clusters, 1.772 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:30:02 : Identified 6 Clusters, 2.016 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:30:02 : Saving LSI Iteration, 2.016 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:30:16 : Creating Cluster Matrix on the total Group Features, 2.247 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:30:30 : Computing Variable Features, 2.474 mins elapsed.</span></code></pre>
<pre><code><span class="co">## ###########</span>
<span class="co">## 2024-10-23 12:30:30 : Running LSI (2 of 2) on Variable Features, 2.477 mins elapsed.</span>
<span class="co">## ###########</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:30:30 : Creating Partial Matrix, 2.477 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:30:59 : Computing LSI, 2.964 mins elapsed.</span></code></pre>
<pre><code><span class="co">## 2024-10-23 12:31:22 : Finished Running IterativeLSI, 3.345 mins elapsed.</span></code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">RNA</span>, assay<span class="op">=</span><span class="st">"RNA"</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">object</span>, normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>, scale.factor <span class="op">=</span> <span class="fl">10000</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">object</span>, nfeatures<span class="op">=</span><span class="fl">3000</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">object</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">object</span>, npcs<span class="op">=</span><span class="fl">50</span>, reduction.name<span class="op">=</span><span class="st">"RNA_PCA"</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>

<span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/setDimRed.html">setDimRed</a></span><span class="op">(</span><span class="va">proj</span>, mtx<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Embeddings.html" class="external-link">Embeddings</a></span><span class="op">(</span><span class="va">object</span><span class="op">[[</span><span class="st">"RNA_PCA"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, type<span class="op">=</span><span class="st">"reducedDims"</span>, reduction.name<span class="op">=</span><span class="st">"RNA_PCA"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/ArchR/man/saveArchRProject.html" class="external-link">saveArchRProject</a></span><span class="op">(</span><span class="va">proj</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Saving ArchRProject...</span></code></pre>
<pre><code><span class="co">## Loading ArchRProject...</span></code></pre>
<pre><code><span class="co">## Successfully loaded ArchRProject!</span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">##                                                    / |</span>
<span class="co">##                                                  /    \</span>
<span class="co">##             .                                  /      |.</span>
<span class="co">##             \\\                              /        |.</span>
<span class="co">##               \\\                          /           `|.</span>
<span class="co">##                 \\\                      /              |.</span>
<span class="co">##                   \                    /                |\</span>
<span class="co">##                   \\#####\           /                  ||</span>
<span class="co">##                 ==###########&gt;      /                   ||</span>
<span class="co">##                  \\##==......\    /                     ||</span>
<span class="co">##             ______ =       =|__ /__                     ||      \\\</span>
<span class="co">##         ,--' ,----`-,__ ___/'  --,-`-===================##========&gt;</span>
<span class="co">##        \               '        ##_______ _____ ,--,__,=##,__   ///</span>
<span class="co">##         ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/</span>
<span class="co">##         -,____,---'       \\####\\________________,--\\_##,/</span>
<span class="co">##            ___      .______        ______  __    __  .______      </span>
<span class="co">##           /   \     |   _  \      /      ||  |  |  | |   _  \     </span>
<span class="co">##          /  ^  \    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    </span>
<span class="co">##         /  /_\  \   |      /     |  |     |   __   | |      /     </span>
<span class="co">##        /  _____  \  |  |\  \\___ |  `----.|  |  |  | |  |\  \\___.</span>
<span class="co">##       /__/     \__\ | _| `._____| \______||__|  |__| | _| `._____|</span>
<span class="co">## </span></code></pre>
<pre><code><span class="co">## </span>
<span class="co">##            ___      .______        ______  __    __  .______      </span>
<span class="co">##           /   \     |   _  \      /      ||  |  |  | |   _  \     </span>
<span class="co">##          /  ^  \    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    </span>
<span class="co">##         /  /_\  \   |      /     |  |     |   __   | |      /     </span>
<span class="co">##        /  _____  \  |  |\  \\___ |  `----.|  |  |  | |  |\  \\___.</span>
<span class="co">##       /__/     \__\ | _| `._____| \______||__|  |__| | _| `._____|</span>
<span class="co">## </span></code></pre>
<pre><code><span class="co">## class: ArchRProject </span>
<span class="co">## outputDirectory: /data/sz753404/git_code/MOJITOO/vignettes/PBMC </span>
<span class="co">## samples(1): pbmc</span>
<span class="co">## sampleColData names(1): ArrowFiles</span>
<span class="co">## cellColData names(20): Sample TSSEnrichment ... Gex_MitoRatio</span>
<span class="co">##   Gex_RiboRatio</span>
<span class="co">## numberOfCells(1): 11763</span>
<span class="co">## medianTSS(1): 13.659</span>
<span class="co">## medianFrags(1): 13486</span></code></pre>
</div>
<div class="section level3">
<h3 id="run-mojitoo">5. RUN MOJITOO<a class="anchor" aria-label="anchor" href="#run-mojitoo"></a>
</h3>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ArchR/man/loadArchRProject.html" class="external-link">loadArchRProject</a></span><span class="op">(</span><span class="st">"PBMC"</span>, showLogo<span class="op">=</span><span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Successfully loaded ArchRProject!</span></code></pre>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mojitoo.html">mojitoo</a></span><span class="op">(</span>
     object<span class="op">=</span><span class="va">proj</span>,
     reduction.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"RNA_PCA"</span>, <span class="st">"IterativeLSI"</span><span class="op">)</span>,
     dims.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">30</span><span class="op">)</span>,
     is.reduction.center<span class="op">=</span><span class="cn">T</span>,
     is.reduction.scale<span class="op">=</span><span class="cn">T</span>,
     reduction.name<span class="op">=</span><span class="st">'MOJITOO'</span>
<span class="op">)</span></code></pre></div>
<pre><code><span class="co">## processing RNA_PCA</span></code></pre>
<pre><code><span class="co">## adding IterativeLSI</span></code></pre>
<pre><code><span class="co">## 1 round cc 29</span></code></pre>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">proj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ArchR/man/addUMAP.html" class="external-link">addUMAP</a></span><span class="op">(</span>
    ArchRProj <span class="op">=</span> <span class="va">proj</span>,
    reducedDims <span class="op">=</span> <span class="st">"MOJITOO"</span>,
    name <span class="op">=</span> <span class="st">"MOJITOO_UMAP"</span>,
    nNeighbors <span class="op">=</span> <span class="fl">30</span>,
    minDist <span class="op">=</span> <span class="fl">0.5</span>,
    metric <span class="op">=</span> <span class="st">"cosine"</span>
<span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 12:31:47 UMAP embedding parameters a = 0.583 b = 1.334</span></code></pre>
<pre><code><span class="co">## 12:31:47 Read 11763 rows and found 29 numeric columns</span></code></pre>
<pre><code><span class="co">## 12:31:47 Using Annoy for neighbor search, n_neighbors = 30</span></code></pre>
<pre><code><span class="co">## 12:31:47 Building Annoy index with metric = cosine, n_trees = 50</span></code></pre>
<pre><code><span class="co">## 0%   10   20   30   40   50   60   70   80   90   100%</span></code></pre>
<pre><code><span class="co">## [----|----|----|----|----|----|----|----|----|----|</span></code></pre>
<pre><code><span class="co">## **************************************************|</span>
<span class="co">## 12:31:48 Writing NN index file to temp file /tmp/Rtmpbwz0TN/filefa26e6f841aa7</span>
<span class="co">## 12:31:48 Searching Annoy index using 64 threads, search_k = 3000</span>
<span class="co">## 12:31:48 Annoy recall = 100%</span>
<span class="co">## 12:31:49 Commencing smooth kNN distance calibration using 64 threads with target n_neighbors = 30</span>
<span class="co">## 12:31:51 Initializing from normalized Laplacian + noise (using irlba)</span>
<span class="co">## 12:31:51 Commencing optimization for 200 epochs, with 461100 positive edges</span>
<span class="co">## 12:31:57 Optimization finished</span>
<span class="co">## 12:31:57 Creating temp model dir /tmp/Rtmpbwz0TN/dirfa26e2260917e</span>
<span class="co">## 12:31:57 Creating dir /tmp/Rtmpbwz0TN/dirfa26e2260917e</span>
<span class="co">## 12:31:58 Changing to /tmp/Rtmpbwz0TN/dirfa26e2260917e</span>
<span class="co">## 12:31:58 Creating /data/sz753404/git_code/MOJITOO/vignettes/PBMC/Embeddings/Save-Uwot-UMAP-Params-MOJITOO-fa26e2cf20242-Date-2024-10-23_Time-12-31-57.tar</span></code></pre>
</div>
<div class="section level3">
<h3 id="umap-of-true-labels">6. UMAP of True labels<a class="anchor" aria-label="anchor" href="#umap-of-true-labels"></a>
</h3>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ArchR/man/plotEmbedding.html" class="external-link">plotEmbedding</a></span><span class="op">(</span>ArchRProj <span class="op">=</span> <span class="va">proj</span>, colorBy <span class="op">=</span> <span class="st">"cellColData"</span>, name <span class="op">=</span> <span class="st">"annotation"</span>, embedding <span class="op">=</span> <span class="st">"MOJITOO_UMAP"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ArchR logging to : ArchRLogs/ArchR-plotEmbedding-fa26e2e6ed43f-Date-2024-10-23_Time-12-31-58.log</span>
<span class="co">## If there is an issue, please report to github with logFile!</span></code></pre>
<pre><code><span class="co">## Getting UMAP Embedding</span></code></pre>
<pre><code><span class="co">## ColorBy = cellColData</span></code></pre>
<pre><code><span class="co">## Plotting Embedding</span></code></pre>
<pre><code><span class="co">## 1 </span>
<span class="co">## ArchR logging successful to : ArchRLogs/ArchR-plotEmbedding-fa26e2e6ed43f-Date-2024-10-23_Time-12-31-58.log</span></code></pre>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p</span></code></pre></div>
<p><img src="ArchRObject_integration_files/figure-html/UMAP-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="sessioninfo">7. sessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h3>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.0 (2021-05-18)</span>
<span class="co">## Platform: x86_64-conda-linux-gnu (64-bit)</span>
<span class="co">## Running under: Rocky Linux 8.10 (Green Obsidian)</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /data/sz753404/miniconda3/envs/schema/lib/libopenblasp-r0.3.21.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.utf-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.utf-8    </span>
<span class="co">##  [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.utf-8   </span>
<span class="co">##  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">##  [1] grid      stats4    parallel  stats     graphics  grDevices utils    </span>
<span class="co">##  [8] datasets  methods   base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] uwot_0.1.14                       nabor_0.5.0                      </span>
<span class="co">##  [3] gridExtra_2.3                     Rsamtools_2.8.0                  </span>
<span class="co">##  [5] BSgenome.Hsapiens.UCSC.hg38_1.4.3 BSgenome_1.60.0                  </span>
<span class="co">##  [7] rtracklayer_1.52.1                Biostrings_2.62.0                </span>
<span class="co">##  [9] XVector_0.34.0                    EnsDb.Hsapiens.v86_2.99.0        </span>
<span class="co">## [11] ensembldb_2.16.4                  AnnotationFilter_1.16.0          </span>
<span class="co">## [13] GenomicFeatures_1.44.2            AnnotationDbi_1.56.2             </span>
<span class="co">## [15] stringr_1.5.0                     ggsci_2.9                        </span>
<span class="co">## [17] SeuratObject_4.1.3                Seurat_4.3.0                     </span>
<span class="co">## [19] ArchR_1.0.1                       magrittr_2.0.3                   </span>
<span class="co">## [21] rhdf5_2.36.0                      Matrix_1.5-3                     </span>
<span class="co">## [23] data.table_1.14.8                 SummarizedExperiment_1.22.0      </span>
<span class="co">## [25] Biobase_2.54.0                    GenomicRanges_1.44.0             </span>
<span class="co">## [27] GenomeInfoDb_1.30.1               IRanges_2.28.0                   </span>
<span class="co">## [29] S4Vectors_0.32.4                  BiocGenerics_0.40.0              </span>
<span class="co">## [31] MatrixGenerics_1.4.3              matrixStats_0.62.0               </span>
<span class="co">## [33] ggplot2_3.4.1                     MOJITOO_1.0                      </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] rappdirs_0.3.3           scattermore_0.8          ragg_1.3.3              </span>
<span class="co">##   [4] tidyr_1.2.1              bit64_4.0.5              knitr_1.42              </span>
<span class="co">##   [7] irlba_2.3.5.1            DelayedArray_0.18.0      rpart_4.1.19            </span>
<span class="co">##  [10] KEGGREST_1.34.0          RCurl_1.98-1.10          doParallel_1.0.16       </span>
<span class="co">##  [13] generics_0.1.3           cowplot_1.1.1            RSQLite_2.3.0           </span>
<span class="co">##  [16] RANN_2.6.1               future_1.28.0            bit_4.0.5               </span>
<span class="co">##  [19] spatstat.data_3.0-0      xml2_1.3.3               httpuv_1.6.5            </span>
<span class="co">##  [22] assertthat_0.2.1         xfun_0.37                hms_1.1.2               </span>
<span class="co">##  [25] jquerylib_0.1.4          evaluate_0.20            promises_1.2.0.1        </span>
<span class="co">##  [28] fansi_1.0.4              restfulr_0.0.13          progress_1.2.2          </span>
<span class="co">##  [31] dbplyr_2.3.1             igraph_1.4.1             DBI_1.1.3               </span>
<span class="co">##  [34] htmlwidgets_1.6.1        spatstat.geom_3.0-3      purrr_1.0.1             </span>
<span class="co">##  [37] ellipsis_0.3.2           ks_1.13.3                dplyr_1.1.0             </span>
<span class="co">##  [40] backports_1.4.1          biomaRt_2.48.3           deldir_1.0-6            </span>
<span class="co">##  [43] vctrs_0.5.2              Cairo_1.5-15             here_1.0.1              </span>
<span class="co">##  [46] ROCR_1.0-11              abind_1.4-5              withr_2.5.0             </span>
<span class="co">##  [49] cachem_1.0.7             Gviz_1.36.2              progressr_0.10.1        </span>
<span class="co">##  [52] checkmate_2.1.0          sctransform_0.3.5        GenomicAlignments_1.28.0</span>
<span class="co">##  [55] prettyunits_1.1.1        mclust_5.4.9             goftest_1.2-3           </span>
<span class="co">##  [58] cluster_2.1.4            lazyeval_0.2.2           crayon_1.5.2            </span>
<span class="co">##  [61] hdf5r_1.3.5              spatstat.explore_3.0-5   labeling_0.4.2          </span>
<span class="co">##  [64] pkgconfig_2.0.3          nlme_3.1-160             ProtGenerics_1.26.0     </span>
<span class="co">##  [67] nnet_7.3-18              rlang_1.0.6              globals_0.16.1          </span>
<span class="co">##  [70] lifecycle_1.0.3          miniUI_0.1.1.1           filelock_1.0.2          </span>
<span class="co">##  [73] BiocFileCache_2.0.0      dichromat_2.0-0.1        rprojroot_2.0.3         </span>
<span class="co">##  [76] polyclip_1.10-4          lmtest_0.9-40            Rhdf5lib_1.16.0         </span>
<span class="co">##  [79] zoo_1.8-10               base64enc_0.1-3          ggridges_0.5.3          </span>
<span class="co">##  [82] GlobalOptions_0.1.2      png_0.1-8                viridisLite_0.4.1       </span>
<span class="co">##  [85] rjson_0.2.20             bitops_1.0-7             KernSmooth_2.23-20      </span>
<span class="co">##  [88] rhdf5filters_1.4.0       blob_1.2.3               shape_1.4.6             </span>
<span class="co">##  [91] parallelly_1.32.1        spatstat.random_3.0-1    jpeg_0.1-9              </span>
<span class="co">##  [94] scales_1.2.1             memoise_2.0.1            plyr_1.8.8              </span>
<span class="co">##  [97] ica_1.0-3                zlibbioc_1.40.0          hdrcde_3.4              </span>
<span class="co">## [100] compiler_4.1.0           BiocIO_1.2.0             RColorBrewer_1.1-3      </span>
<span class="co">## [103] clue_0.3-60              fitdistrplus_1.1-8       cli_3.6.0               </span>
<span class="co">## [106] listenv_0.8.0            patchwork_1.1.2          pbapply_1.7-0           </span>
<span class="co">## [109] htmlTable_2.3.0          Formula_1.2-4            MASS_7.3-58.1           </span>
<span class="co">## [112] tidyselect_1.2.0         stringi_1.7.12           textshaping_0.3.6       </span>
<span class="co">## [115] highr_0.10               yaml_2.3.7               latticeExtra_0.6-29     </span>
<span class="co">## [118] ggrepel_0.9.3            sass_0.4.5               VariantAnnotation_1.38.0</span>
<span class="co">## [121] tools_4.1.0              future.apply_1.9.0       circlize_0.4.13         </span>
<span class="co">## [124] rstudioapi_0.14          foreach_1.5.2            foreign_0.8-84          </span>
<span class="co">## [127] farver_2.1.1             Rtsne_0.16               digest_0.6.31           </span>
<span class="co">## [130] pracma_2.3.8             shiny_1.7.2              Rcpp_1.0.10             </span>
<span class="co">## [133] later_1.3.0              RcppAnnoy_0.0.19         fda_5.5.1               </span>
<span class="co">## [136] httr_1.4.5               biovizBase_1.40.0        ComplexHeatmap_2.11.1   </span>
<span class="co">## [139] colorspace_2.1-0         XML_3.99-0.13            fs_1.6.1                </span>
<span class="co">## [142] tensor_1.5               rainbow_3.6              reticulate_1.25         </span>
<span class="co">## [145] splines_4.1.0            spatstat.utils_3.1-0     pkgdown_2.0.3           </span>
<span class="co">## [148] sp_1.5-1                 plotly_4.10.1            systemfonts_1.0.4       </span>
<span class="co">## [151] xtable_1.8-4             jsonlite_1.8.4           fds_1.8                 </span>
<span class="co">## [154] corpcor_1.6.10           R6_2.5.1                 Hmisc_5.0-1             </span>
<span class="co">## [157] ramify_0.3.3             pillar_1.8.1             htmltools_0.5.4         </span>
<span class="co">## [160] mime_0.12                glue_1.6.2               fastmap_1.1.1           </span>
<span class="co">## [163] BiocParallel_1.28.3      deSolve_1.32             codetools_0.2-19        </span>
<span class="co">## [166] pcaPP_2.0-1              mvtnorm_1.1-3            utf8_1.2.3              </span>
<span class="co">## [169] lattice_0.20-45          bslib_0.4.2              spatstat.sparse_3.0-0   </span>
<span class="co">## [172] tibble_3.2.0             curl_5.0.0               leiden_0.4.2            </span>
<span class="co">## [175] gtools_3.9.4             survival_3.5-5           rmarkdown_2.20          </span>
<span class="co">## [178] desc_1.4.2               munsell_0.5.0            GetoptLong_1.0.5        </span>
<span class="co">## [181] GenomeInfoDbData_1.2.7   iterators_1.0.14         reshape2_1.4.4          </span>
<span class="co">## [184] gtable_0.3.1</span></code></pre>
</div>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Mingbo Cheng.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
