<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MOJITOO">
<title>Using MOJITOO in R for Seurat • MOJITOO</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Using MOJITOO in R for Seurat">
<meta property="og:description" content="MOJITOO">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MOJITOO</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/ArchRObject_integration.html">Using MOJITOO in R for ArchR</a>
    <a class="dropdown-item" href="../articles/Matrix_integration.html">Using MOJITOO in R for Matrix input</a>
    <a class="dropdown-item" href="../articles/SeuratObject_integration.html">Using MOJITOO in R for Seurat</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Using MOJITOO in R for Seurat</h1>
                        <h4 data-toc-skip class="author">Mingbo
Cheng</h4>
            
            <h4 data-toc-skip class="date">October 23, 2024</h4>
      
      
      <div class="d-none name"><code>SeuratObject_integration.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="download-data">1. download data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h3>
<pre><code><span class="co">## 2024-10-23 12:32:32 URL:https://cf.10xgenomics.com/samples/cell-arc/1.0.0/pbmc_granulocyte_sorted_10k/pbmc_granulocyte_sorted_10k_filtered_feature_bc_matrix.h5 [162282142/162282142] -&gt; "pbmc_granulocyte_sorted_10k_filtered_feature_bc_matrix.h5.1" [1]</span>
<span class="co">## WARNING: cannot verify costalab.ukaachen.de's certificate, issued by ‘CN=GEANT OV RSA CA 4,O=GEANT Vereniging,C=NL’:</span>
<span class="co">##   Unable to locally verify the issuer's authority.</span>
<span class="co">## 2024-10-23 12:32:32 URL:https://costalab.ukaachen.de/open_data/MOJITOO/PBMC-Multiom_annotation.tsv [400178/400178] -&gt; "PBMC-Multiom_annotation.tsv.1" [1]</span></code></pre>
</div>
<div class="section level3">
<h3 id="library">2. library<a class="anchor" aria-label="anchor" href="#library"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/timoast/signac" class="external-link">Signac</a></span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MOJITOO</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://nanx.me/ggsci/" class="external-link">ggsci</a></span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="create-seurat-object">3. Create Seurat Object<a class="anchor" aria-label="anchor" href="#create-seurat-object"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mtxs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/Read10X_h5.html" class="external-link">Read10X_h5</a></span><span class="op">(</span><span class="st">"pbmc_granulocyte_sorted_10k_filtered_feature_bc_matrix.h5"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Genome matrix has multiple modalities, returning a list of matrices for this genome</span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">RNA</span> <span class="op">&lt;-</span> <span class="va">mtxs</span><span class="op">[[</span><span class="st">"Gene Expression"</span><span class="op">]</span><span class="op">]</span>
<span class="va">ATAC</span> <span class="op">&lt;-</span> <span class="va">mtxs</span><span class="op">[[</span><span class="st">"Peaks"</span><span class="op">]</span><span class="op">]</span>
<span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ATAC</span><span class="op">)</span>
<span class="va">ATAC</span> <span class="op">&lt;-</span> <span class="va">ATAC</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/startsWith.html" class="external-link">startsWith</a></span><span class="op">(</span><span class="va">peaks</span>, <span class="st">"chr"</span><span class="op">)</span><span class="op">)</span>, <span class="op">]</span>

<span class="va">meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"PBMC-Multiom_annotation.tsv"</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span>

<span class="va">RNA</span> <span class="op">&lt;-</span> <span class="va">RNA</span><span class="op">[</span>, <span class="va">meta</span><span class="op">$</span><span class="va">barcode</span><span class="op">]</span>
<span class="va">ATAC</span> <span class="op">&lt;-</span> <span class="va">ATAC</span><span class="op">[</span>, <span class="va">meta</span><span class="op">$</span><span class="va">barcode</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">meta</span><span class="op">$</span><span class="va">barcode</span>
<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">RNA</span>, meta.data<span class="op">=</span><span class="va">meta</span>, assay<span class="op">=</span><span class="st">"RNA"</span><span class="op">)</span>
<span class="va">object</span><span class="op">[[</span><span class="st">"Peaks"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/CreateAssayObject.html" class="external-link">CreateAssayObject</a></span><span class="op">(</span>counts<span class="op">=</span><span class="va">ATAC</span><span class="op">)</span>
<span class="va">object</span><span class="op">$</span><span class="va">celltype</span> <span class="op">&lt;-</span> <span class="va">object</span><span class="op">$</span><span class="va">annotation</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">mtxs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">RNA</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">ATAC</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/gc.html" class="external-link">gc</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##             used   (Mb) gc trigger   (Mb)  max used   (Mb)</span>
<span class="co">## Ncells  12885724  688.2   20530678 1096.5  16201714  865.3</span>
<span class="co">## Vcells 185070328 1412.0  602551575 4597.2 589967819 4501.1</span></code></pre>
</div>
<div class="section level3">
<h3 id="subset-if-need-to-save-memory">subset if need to save memory<a class="anchor" aria-label="anchor" href="#subset-if-need-to-save-memory"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Signac/man/subset.Motif.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">object</span>, cells<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">0.3</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="rna-atac-dimension-reductions">4. RNA &amp; ATAC dimension reductions<a class="anchor" aria-label="anchor" href="#rna-atac-dimension-reductions"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## RNA pre-processing and PCA dimension reduction</span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"RNA"</span>
<span class="va">object</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="va">object</span>, normalization.method <span class="op">=</span> <span class="st">"LogNormalize"</span>, scale.factor <span class="op">=</span> <span class="fl">10000</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="va">object</span>, nfeatures<span class="op">=</span><span class="fl">3000</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="va">object</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="va">object</span>, npcs<span class="op">=</span><span class="fl">50</span>, reduction.name<span class="op">=</span><span class="st">"RNA_PCA"</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>

<span class="co">## RNA pre-processing and LSI dimension reduction</span>
<span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Peaks"</span>
<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Signac/man/RunTFIDF.html" class="external-link">RunTFIDF</a></span><span class="op">(</span><span class="va">object</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning in RunTFIDF.default(object = GetAssayData(object = object, slot =</span>
<span class="co">## "counts"), : Some features contain 0 total counts</span></code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Signac/man/FindTopFeatures.html" class="external-link">FindTopFeatures</a></span><span class="op">(</span><span class="va">object</span>, min.cutoff <span class="op">=</span> <span class="st">'q0'</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Signac/man/RunSVD.html" class="external-link">RunSVD</a></span><span class="op">(</span><span class="va">object</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="run-mojitoo">5. RUN MOJITOO<a class="anchor" aria-label="anchor" href="#run-mojitoo"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mojitoo.html">mojitoo</a></span><span class="op">(</span>
     object<span class="op">=</span><span class="va">object</span>,
     reduction.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"RNA_PCA"</span>, <span class="st">"lsi"</span><span class="op">)</span>,
     dims.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">50</span><span class="op">)</span>, <span class="co">## exclude 1st dimension of LSI</span>
     reduction.name<span class="op">=</span><span class="st">'MOJITOO'</span>,
     assay<span class="op">=</span><span class="st">"RNA"</span>
<span class="op">)</span></code></pre></div>
<pre><code><span class="co">## processing RNA_PCA</span></code></pre>
<pre><code><span class="co">## adding lsi</span></code></pre>
<pre><code><span class="co">## 1 round cc 44</span></code></pre>
<pre><code><span class="co">## Warning: Keys should be one or more alphanumeric characters followed by an</span>
<span class="co">## underscore, setting key from MOJITOO to MOJITOO_</span></code></pre>
<pre><code><span class="co">## Warning: All keys should be one or more alphanumeric characters followed by an</span>
<span class="co">## underscore '_', setting key to MOJITOO_</span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/DefaultAssay.html" class="external-link">DefaultAssay</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"RNA"</span>
<span class="va">embedd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/Embeddings.html" class="external-link">Embeddings</a></span><span class="op">(</span><span class="va">object</span><span class="op">[[</span><span class="st">"MOJITOO"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span><span class="va">object</span>, reduction<span class="op">=</span><span class="st">"MOJITOO"</span>, reduction.name<span class="op">=</span><span class="st">"MOJITOO_UMAP"</span>, dims<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">embedd</span><span class="op">)</span>, verbose<span class="op">=</span><span class="cn">F</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric</span>
<span class="co">## To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'</span>
<span class="co">## This message will be shown once per session</span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">object</span>, <span class="st">"seu.Rds"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="umap-of-true-labels">6. UMAP of True labels<a class="anchor" aria-label="anchor" href="#umap-of-true-labels"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">object</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"seu.Rds"</span><span class="op">)</span> <span class="co">## 1/3 cells</span>
<span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">object</span>, reduction <span class="op">=</span> <span class="st">"MOJITOO_UMAP"</span>, group.by <span class="op">=</span> <span class="st">"celltype"</span>, cols<span class="op">=</span><span class="fu">ggsci</span><span class="fu">::</span><span class="fu"><a href="https://nanx.me/ggsci/reference/pal_igv.html" class="external-link">pal_igv</a></span><span class="op">(</span><span class="op">)</span><span class="op">(</span><span class="fl">30</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="SeuratObject_integration_files/figure-html/UMAPPlot-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="mojitoo-ccs">7. MOJITOO CCs<a class="anchor" aria-label="anchor" href="#mojitoo-ccs"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GeneCCDimPlot.html">GeneCCDimPlot</a></span><span class="op">(</span><span class="va">object</span>,
              CCsToPlot <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span>,
              RNA.assay<span class="op">=</span><span class="st">"RNA"</span>,
              umap <span class="op">=</span> <span class="st">"MOJITOO_UMAP"</span>,
              MOJITOO.reduction<span class="op">=</span><span class="st">"MOJITOO"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## adding MOJITOO assay...</span></code></pre>
<pre><code><span class="co">## Warning: Keys should be one or more alphanumeric characters followed by an</span>
<span class="co">## underscore, setting key from mojitoo.assay_ to mojitooassay_</span></code></pre>
<pre><code><span class="co">## Centering and scaling data matrix</span></code></pre>
<pre><code><span class="co">## Scale for <span style="color: #00BB00;">colour</span> is already present.</span>
<span class="co">## Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span>
<span class="co">## Scale for <span style="color: #00BB00;">colour</span> is already present.</span>
<span class="co">## Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span>
<span class="co">## Scale for <span style="color: #00BB00;">colour</span> is already present.</span>
<span class="co">## Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span>
<span class="co">## Scale for <span style="color: #00BB00;">colour</span> is already present.</span>
<span class="co">## Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></code></pre>
<p><img src="SeuratObject_integration_files/figure-html/CCDimPlot-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="heatmap">Heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/GeneCCHeatmap.Seurat.html">GeneCCHeatmap</a></span><span class="op">(</span><span class="va">object</span>,
                    CCsToPlot <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,
                    RNA.assay<span class="op">=</span><span class="st">"RNA"</span>,
                    colorbar.group <span class="op">=</span> <span class="st">"celltype"</span>,
                    MOJITOO.reduction<span class="op">=</span><span class="st">"MOJITOO"</span>,
                    filter.mito <span class="op">=</span> <span class="cn">T</span>,
                    filter.ribo <span class="op">=</span> <span class="cn">T</span>,
                    topN <span class="op">=</span> <span class="fl">10</span>
                    <span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ccsybygenes...</span></code></pre>
<pre><code><span class="co">## Centering and scaling data matrix</span></code></pre>
<pre><code><span class="co">## $`1`</span></code></pre>
<p><img src="SeuratObject_integration_files/figure-html/CCHeatmap-1.png" width="700"></p>
<pre><code><span class="co">## </span>
<span class="co">## $`2`</span></code></pre>
<p><img src="SeuratObject_integration_files/figure-html/CCHeatmap-2.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="download-track-dependent-data">Download track dependent data<a class="anchor" aria-label="anchor" href="#download-track-dependent-data"></a>
</h3>
<pre><code><span class="co">## WARNING: cannot verify costalab.ukaachen.de's certificate, issued by ‘CN=GEANT OV RSA CA 4,O=GEANT Vereniging,C=NL’:</span>
<span class="co">##   Unable to locally verify the issuer's authority.</span>
<span class="co">## 2024-10-23 12:35:28 URL:https://costalab.ukaachen.de/open_data/MOJITOO/genes.gtf.zip [24022019/24022019] -&gt; "genes.gtf.zip" [1]</span>
<span class="co">## Archive:  genes.gtf.zip</span>
<span class="co">##   inflating: genes.gtf               </span>
<span class="co">## WARNING: cannot verify costalab.ukaachen.de's certificate, issued by ‘CN=GEANT OV RSA CA 4,O=GEANT Vereniging,C=NL’:</span>
<span class="co">##   Unable to locally verify the issuer's authority.</span>
<span class="co">## 2024-10-23 12:35:37 URL:https://costalab.ukaachen.de/open_data/MOJITOO/bigwig.zip [717556067/717556067] -&gt; "bigwig.zip" [1]</span>
<span class="co">## Archive:  bigwig.zip</span>
<span class="co">##    creating: bigwig/</span>
<span class="co">##   inflating: bigwig/CD14+_Monocytes.bw  </span>
<span class="co">##   inflating: bigwig/CD4_Memory.bw    </span>
<span class="co">##   inflating: bigwig/Platelets.bw     </span>
<span class="co">##   inflating: bigwig/CD4_Naive.bw     </span>
<span class="co">##   inflating: bigwig/CD16+_Monocytes.bw  </span>
<span class="co">##   inflating: bigwig/pre-B_cell.bw    </span>
<span class="co">##   inflating: bigwig/CD8_Naive.bw     </span>
<span class="co">##   inflating: bigwig/pDC.bw           </span>
<span class="co">##   inflating: bigwig/Dendritic_cell.bw  </span>
<span class="co">##   inflating: bigwig/NK_cell.bw       </span>
<span class="co">##   inflating: bigwig/Double_negative_T_cell.bw  </span>
<span class="co">##   inflating: bigwig/CD8_effector.bw  </span>
<span class="co">##   inflating: bigwig/B_cell_progenitor.bw</span></code></pre>
</div>
<div class="section level3">
<h3 id="tracks">Tracks<a class="anchor" aria-label="anchor" href="#tracks"></a>
</h3>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data_track_bws</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
    <span class="st">"B cell progenitor"</span>             <span class="op">=</span> <span class="st">"bigwig/B_cell_progenitor.bw"</span>,
    <span class="st">"CD14+ Monocytes"</span>               <span class="op">=</span> <span class="st">"bigwig/CD14+_Monocytes.bw"</span>,
    <span class="st">"CD16+ Monocytes"</span>               <span class="op">=</span> <span class="st">"bigwig/CD16+_Monocytes.bw"</span>,
    <span class="st">"CD4 Memory"</span>                    <span class="op">=</span> <span class="st">"bigwig/CD4_Memory.bw"</span>,
    <span class="st">"CD4 Naive"</span>                     <span class="op">=</span> <span class="st">"bigwig/CD4_Naive.bw"</span>,
    <span class="st">"CD8 effector"</span>                  <span class="op">=</span> <span class="st">"bigwig/CD8_effector.bw"</span>,
    <span class="st">"CD8 Naive"</span>                     <span class="op">=</span> <span class="st">"bigwig/CD8_Naive.bw"</span>,
    <span class="st">"Dendritic cell"</span>                <span class="op">=</span> <span class="st">"bigwig/Dendritic_cell.bw"</span>,
    <span class="st">"Double negative T cell"</span>        <span class="op">=</span> <span class="st">"bigwig/Double_negative_T_cell.bw"</span>,
    <span class="st">"NK cell"</span>                       <span class="op">=</span> <span class="st">"bigwig/NK_cell.bw"</span>,
    <span class="st">"pDC"</span>                           <span class="op">=</span> <span class="st">"bigwig/pDC.bw"</span>,
    <span class="st">"Platelets"</span>                     <span class="op">=</span> <span class="st">"bigwig/Platelets.bw"</span>,
    <span class="st">"pre-B cell"</span>                    <span class="op">=</span> <span class="st">"bigwig/pre-B_cell.bw"</span>
<span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: package 'BiocGenerics' was built under R version 4.1.1</span></code></pre>
<pre><code><span class="co">## Warning: package 'S4Vectors' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## Warning: package 'IRanges' was built under R version 4.1.3</span></code></pre>
<pre><code><span class="co">## Warning: package 'GenomeInfoDb' was built under R version 4.1.3</span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/readGFF.html" class="external-link">readGFF</a></span><span class="op">(</span><span class="st">"genes.gtf"</span><span class="op">)</span>
<span class="va">gene_model</span><span class="op">$</span><span class="va">chromosome</span> <span class="op">&lt;-</span> <span class="va">gene_model</span><span class="op">$</span><span class="va">seqid</span>
<span class="va">gene_model</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="va">gene_model</span><span class="op">$</span><span class="va">gene_id</span>
<span class="va">gene_model</span><span class="op">$</span><span class="va">transcript</span> <span class="op">&lt;-</span> <span class="va">gene_model</span><span class="op">$</span><span class="va">transcript_id</span>
<span class="va">gene_model</span><span class="op">$</span><span class="va">symbol</span> <span class="op">&lt;-</span> <span class="va">gene_model</span><span class="op">$</span><span class="va">gene_name</span>
<span class="va">gene_model</span><span class="op">$</span><span class="va">exon</span> <span class="op">&lt;-</span> <span class="va">gene_model</span><span class="op">$</span><span class="va">exon_id</span>
<span class="va">gene_model</span><span class="op">$</span><span class="va">width</span> <span class="op">&lt;-</span> <span class="va">gene_model</span><span class="op">$</span><span class="va">end</span> <span class="op">-</span> <span class="va">gene_model</span><span class="op">$</span><span class="va">start</span> <span class="op">+</span> <span class="fl">1</span>
<span class="va">gene_model</span><span class="op">$</span><span class="va">feature</span> <span class="op">&lt;-</span> <span class="va">gene_model</span><span class="op">$</span><span class="va">transcript_type</span>
<span class="va">gene_model</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Signac/man/subset.Motif.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">gene_model</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">transcript</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">exon</span><span class="op">)</span><span class="op">)</span>

<span class="va">gtree</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ATACTrack.html">ATACTrack</a></span><span class="op">(</span><span class="va">object</span>,
                 CC <span class="op">=</span> <span class="fl">1</span>,
                 group.by<span class="op">=</span><span class="st">"celltype"</span>,
                 bigwig.file.list<span class="op">=</span><span class="va">data_track_bws</span>,
                 MOJITOO.reduction<span class="op">=</span><span class="st">"MOJITOO"</span>,
                 Peak.assay<span class="op">=</span><span class="st">"Peaks"</span>,
                 gene.model<span class="op">=</span><span class="va">gene_model</span>,
                 cols <span class="op">=</span><span class="fu">ggsci</span><span class="fu">::</span><span class="fu"><a href="https://nanx.me/ggsci/reference/pal_igv.html" class="external-link">pal_igv</a></span><span class="op">(</span><span class="op">)</span><span class="op">(</span><span class="fl">51</span><span class="op">)</span>,
                 ylim.datatrack<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">16</span><span class="op">)</span>,
                 fontsize.geneAxis<span class="op">=</span><span class="fl">5</span>,
                 fontsize.geneRegion<span class="op">=</span><span class="fl">10</span>,
                 fontsize.datatrack<span class="op">=</span><span class="fl">8</span>,
                 show.legend<span class="op">=</span><span class="cn">T</span>,
                 genome<span class="op">=</span><span class="st">"hg38"</span>
                 <span class="op">)</span></code></pre></div>
<pre><code><span class="co">## No peaks input, will get top2 posi&amp;nega peaks from CC1</span></code></pre>
<pre><code><span class="co">## loading data track bigwig files</span></code></pre>
<pre><code><span class="co">## column</span></code></pre>
<pre><code><span class="co">##  1</span></code></pre>
<pre><code><span class="co">##  2</span></code></pre>
<pre><code><span class="co">##  3</span></code></pre>
<pre><code><span class="co">##  4</span></code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#grid::grid.newpage()</span>
<span class="fu">grid</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grid/grid.draw.html" class="external-link">grid.draw</a></span><span class="op">(</span><span class="va">gtree</span><span class="op">)</span></code></pre></div>
<p><img src="SeuratObject_integration_files/figure-html/CCTracks-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h3>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.0 (2021-05-18)</span>
<span class="co">## Platform: x86_64-conda-linux-gnu (64-bit)</span>
<span class="co">## Running under: Rocky Linux 8.10 (Green Obsidian)</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS/LAPACK: /data/sz753404/miniconda3/envs/schema/lib/libopenblasp-r0.3.21.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.utf-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.utf-8    </span>
<span class="co">##  [5] LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=en_US.utf-8   </span>
<span class="co">##  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] rtracklayer_1.52.1   GenomicRanges_1.44.0 GenomeInfoDb_1.30.1 </span>
<span class="co">##  [4] IRanges_2.28.0       S4Vectors_0.32.4     BiocGenerics_0.40.0 </span>
<span class="co">##  [7] ggsci_2.9            MOJITOO_1.0          Signac_1.6.0        </span>
<span class="co">## [10] SeuratObject_4.1.3   Seurat_4.3.0        </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] rappdirs_0.3.3              SnowballC_0.7.0            </span>
<span class="co">##   [3] scattermore_0.8             ragg_1.3.3                 </span>
<span class="co">##   [5] tidyr_1.2.1                 ggplot2_3.4.1              </span>
<span class="co">##   [7] bit64_4.0.5                 knitr_1.42                 </span>
<span class="co">##   [9] irlba_2.3.5.1               DelayedArray_0.18.0        </span>
<span class="co">##  [11] rpart_4.1.19                data.table_1.14.8          </span>
<span class="co">##  [13] AnnotationFilter_1.16.0     KEGGREST_1.34.0            </span>
<span class="co">##  [15] RCurl_1.98-1.10             doParallel_1.0.16          </span>
<span class="co">##  [17] generics_0.1.3              GenomicFeatures_1.44.2     </span>
<span class="co">##  [19] cowplot_1.1.1               RSQLite_2.3.0              </span>
<span class="co">##  [21] RANN_2.6.1                  future_1.28.0              </span>
<span class="co">##  [23] bit_4.0.5                   spatstat.data_3.0-0        </span>
<span class="co">##  [25] xml2_1.3.3                  httpuv_1.6.5               </span>
<span class="co">##  [27] assertthat_0.2.1            SummarizedExperiment_1.22.0</span>
<span class="co">##  [29] xfun_0.37                   hms_1.1.2                  </span>
<span class="co">##  [31] jquerylib_0.1.4             evaluate_0.20              </span>
<span class="co">##  [33] promises_1.2.0.1            fansi_1.0.4                </span>
<span class="co">##  [35] restfulr_0.0.13             progress_1.2.2             </span>
<span class="co">##  [37] dbplyr_2.3.1                igraph_1.4.1               </span>
<span class="co">##  [39] DBI_1.1.3                   htmlwidgets_1.6.1          </span>
<span class="co">##  [41] sparsesvd_0.2               spatstat.geom_3.0-3        </span>
<span class="co">##  [43] purrr_1.0.1                 ellipsis_0.3.2             </span>
<span class="co">##  [45] ks_1.13.3                   backports_1.4.1            </span>
<span class="co">##  [47] dplyr_1.1.0                 biomaRt_2.48.3             </span>
<span class="co">##  [49] deldir_1.0-6                MatrixGenerics_1.4.3       </span>
<span class="co">##  [51] vctrs_0.5.2                 Biobase_2.54.0             </span>
<span class="co">##  [53] here_1.0.1                  ensembldb_2.16.4           </span>
<span class="co">##  [55] ROCR_1.0-11                 abind_1.4-5                </span>
<span class="co">##  [57] cachem_1.0.7                withr_2.5.0                </span>
<span class="co">##  [59] ggforce_0.4.1               Gviz_1.36.2                </span>
<span class="co">##  [61] BSgenome_1.60.0             progressr_0.10.1           </span>
<span class="co">##  [63] checkmate_2.1.0             sctransform_0.3.5          </span>
<span class="co">##  [65] GenomicAlignments_1.28.0    prettyunits_1.1.1          </span>
<span class="co">##  [67] mclust_5.4.9                goftest_1.2-3              </span>
<span class="co">##  [69] cluster_2.1.4               lazyeval_0.2.2             </span>
<span class="co">##  [71] crayon_1.5.2                hdf5r_1.3.5                </span>
<span class="co">##  [73] spatstat.explore_3.0-5      labeling_0.4.2             </span>
<span class="co">##  [75] pkgconfig_2.0.3             slam_0.1-49                </span>
<span class="co">##  [77] tweenr_2.0.2                ProtGenerics_1.26.0        </span>
<span class="co">##  [79] nlme_3.1-160                nnet_7.3-18                </span>
<span class="co">##  [81] rlang_1.0.6                 globals_0.16.1             </span>
<span class="co">##  [83] lifecycle_1.0.3             miniUI_0.1.1.1             </span>
<span class="co">##  [85] filelock_1.0.2              BiocFileCache_2.0.0        </span>
<span class="co">##  [87] dichromat_2.0-0.1           rprojroot_2.0.3            </span>
<span class="co">##  [89] polyclip_1.10-4             matrixStats_0.62.0         </span>
<span class="co">##  [91] lmtest_0.9-40               Matrix_1.5-3               </span>
<span class="co">##  [93] ggseqlogo_0.1               Rhdf5lib_1.16.0            </span>
<span class="co">##  [95] zoo_1.8-10                  base64enc_0.1-3            </span>
<span class="co">##  [97] ggridges_0.5.3              GlobalOptions_0.1.2        </span>
<span class="co">##  [99] png_0.1-8                   viridisLite_0.4.1          </span>
<span class="co">## [101] rjson_0.2.20                bitops_1.0-7               </span>
<span class="co">## [103] KernSmooth_2.23-20          rhdf5filters_1.4.0         </span>
<span class="co">## [105] Biostrings_2.62.0           blob_1.2.3                 </span>
<span class="co">## [107] shape_1.4.6                 stringr_1.5.0              </span>
<span class="co">## [109] parallelly_1.32.1           spatstat.random_3.0-1      </span>
<span class="co">## [111] jpeg_0.1-9                  scales_1.2.1               </span>
<span class="co">## [113] memoise_2.0.1               magrittr_2.0.3             </span>
<span class="co">## [115] plyr_1.8.8                  ica_1.0-3                  </span>
<span class="co">## [117] zlibbioc_1.40.0             hdrcde_3.4                 </span>
<span class="co">## [119] compiler_4.1.0              BiocIO_1.2.0               </span>
<span class="co">## [121] RColorBrewer_1.1-3          clue_0.3-60                </span>
<span class="co">## [123] fitdistrplus_1.1-8          Rsamtools_2.8.0            </span>
<span class="co">## [125] cli_3.6.0                   XVector_0.34.0             </span>
<span class="co">## [127] listenv_0.8.0               patchwork_1.1.2            </span>
<span class="co">## [129] pbapply_1.7-0               htmlTable_2.3.0            </span>
<span class="co">## [131] Formula_1.2-4               MASS_7.3-58.1              </span>
<span class="co">## [133] tidyselect_1.2.0            stringi_1.7.12             </span>
<span class="co">## [135] textshaping_0.3.6           highr_0.10                 </span>
<span class="co">## [137] yaml_2.3.7                  latticeExtra_0.6-29        </span>
<span class="co">## [139] ggrepel_0.9.3               grid_4.1.0                 </span>
<span class="co">## [141] VariantAnnotation_1.38.0    sass_0.4.5                 </span>
<span class="co">## [143] fastmatch_1.1-3             tools_4.1.0                </span>
<span class="co">## [145] future.apply_1.9.0          parallel_4.1.0             </span>
<span class="co">## [147] rstudioapi_0.14             circlize_0.4.13            </span>
<span class="co">## [149] foreign_0.8-84              foreach_1.5.2              </span>
<span class="co">## [151] lsa_0.73.3                  gridExtra_2.3              </span>
<span class="co">## [153] farver_2.1.1                Rtsne_0.16                 </span>
<span class="co">## [155] digest_0.6.31               pracma_2.3.8               </span>
<span class="co">## [157] shiny_1.7.2                 qlcMatrix_0.9.7            </span>
<span class="co">## [159] Rcpp_1.0.10                 later_1.3.0                </span>
<span class="co">## [161] fda_5.5.1                   RcppAnnoy_0.0.19           </span>
<span class="co">## [163] httr_1.4.5                  AnnotationDbi_1.56.2       </span>
<span class="co">## [165] biovizBase_1.40.0           ComplexHeatmap_2.11.1      </span>
<span class="co">## [167] colorspace_2.1-0            rainbow_3.6                </span>
<span class="co">## [169] XML_3.99-0.13               fs_1.6.1                   </span>
<span class="co">## [171] tensor_1.5                  reticulate_1.25            </span>
<span class="co">## [173] splines_4.1.0               uwot_0.1.14                </span>
<span class="co">## [175] RcppRoll_0.3.0              spatstat.utils_3.1-0       </span>
<span class="co">## [177] pkgdown_2.0.3               sp_1.5-1                   </span>
<span class="co">## [179] ArchR_1.0.1                 plotly_4.10.1              </span>
<span class="co">## [181] systemfonts_1.0.4           xtable_1.8-4               </span>
<span class="co">## [183] fds_1.8                     jsonlite_1.8.4             </span>
<span class="co">## [185] corpcor_1.6.10              R6_2.5.1                   </span>
<span class="co">## [187] Hmisc_5.0-1                 ramify_0.3.3               </span>
<span class="co">## [189] pillar_1.8.1                htmltools_0.5.4            </span>
<span class="co">## [191] mime_0.12                   glue_1.6.2                 </span>
<span class="co">## [193] fastmap_1.1.1               BiocParallel_1.28.3        </span>
<span class="co">## [195] deSolve_1.32                codetools_0.2-19           </span>
<span class="co">## [197] pcaPP_2.0-1                 mvtnorm_1.1-3              </span>
<span class="co">## [199] utf8_1.2.3                  lattice_0.20-45            </span>
<span class="co">## [201] bslib_0.4.2                 spatstat.sparse_3.0-0      </span>
<span class="co">## [203] tibble_3.2.0                curl_5.0.0                 </span>
<span class="co">## [205] leiden_0.4.2                magick_2.7.3               </span>
<span class="co">## [207] survival_3.5-5              rmarkdown_2.20             </span>
<span class="co">## [209] docopt_0.7.1                desc_1.4.2                 </span>
<span class="co">## [211] munsell_0.5.0               GetoptLong_1.0.5           </span>
<span class="co">## [213] rhdf5_2.36.0                GenomeInfoDbData_1.2.7     </span>
<span class="co">## [215] iterators_1.0.14            reshape2_1.4.4             </span>
<span class="co">## [217] gtable_0.3.1</span></code></pre>
</div>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Mingbo Cheng.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
